Budget Manager/task/src/budget/BudgetApp.java

```diff
@@ -2,100 +2,98 @@
-
-        switch (userInput) {
-            case "1":
-                processIncome(askForIncome());
-                break;
-            case "2":
-                askPurchase();
-                break;
-            case "3":
-                printAllPurchasesAndTotalPrice();
-                break;
-            case "4":
-                printBalance();
-                break;
-            case "5":
-                SavePurchases savePurchases = new SavePurchases();
-                savePurchases.savePurchases(purchaseStore);
-                print("\nPurchases were saved!\n\n");
-                break;
-            case "6":
-                LoadPurchases loadPurchases = new LoadPurchases();
-                this.purchaseStore = loadPurchases.loadPurchases();
-                print("\nPurchases were loaded!\n\n");
-                break;
-            case "0":
-                exitBudgetApp();
-                break;
-            default:
-                break;
+    public void print(ArrayList<String> result) {
+        if (result != null) {
+            for (String string : result) {
+                print(string);
+            }
         }
     }
 
-    private void exitBudgetApp() {
-        print("\nBye!\n\n");
-        isOnline = false;
+    public void run() {
+        boolean isOnline = true;
+        while (isOnline) {
+            switch (Menu.getOptionFromGeneralMenu()) {
+                case "1": // 1) Add income
+                    processIncome(askForIncome());
+                    break;
+                case "2": // 2) Add purchase
+                    askPurchase();
+                    break;
+                case "3": // 3) Show list of purchases
+                    printAllPurchasesAndTotalPrice();
+                    break;
+                case "4": // 4) Balance
+                    printBalance();
+                    break;
+                case "5": // 5) Save
+                    SavePurchases.savePurchases(purchaseStore);
+                    print("\nPurchases were saved!\n\n");
+                    break;
+                case "6": // 6) Load
+                    this.purchaseStore = LoadPurchases.loadPurchases();
+                    print("\nPurchases were loaded!\n\n");
+                    break;
+                case "7": // 7) Analyze (Sort)
+                    processSorting();
+                    break;
+                case "0": // 0) Exit
+                    print("\nBye!\n\n");
+                    isOnline = false;
+                    break;
+                default:
+                    break;
+            }
+        }
     }
 
-    boolean isOnline() {
-        return isOnline;
+    private void processSorting() {
+        if (purchaseStore.getPurchaseStore().isEmpty()) {
+            print("\nPurchase list is empty\n\n");
+            return;
+        }
+
+        boolean goBack = false;
+        while (!goBack) {
+            switch (Menu.getOptionFromSortMenu()) {
```

Comment:
Mis de default case in dit geval, mocht er iets worden toegevoegd aan het programma en is hier nog niet geimplementeerd wil je hoe dan ook dat er een default afgaat.